Data/Aeson/Types/Internal.hs

```diff
@@ -458,7 +459,12 @@ parseEither m v = runParser (m v) [] onError Right
 -- | Annotate an error message with a
 -- <http://goessner.net/articles/JsonPath/ JSONPath> error location.
 formatError :: JSONPath -> String -> String
-formatError path msg = "Error in " ++ format "$" path ++ ": " ++ msg
+formatError path msg = "Error in " ++ formatPath "$" path ++ ": " ++ msg
+
+-- | Format a <http://goessner.net/articles/JsonPath/ JSONPath> as a 'String'.
+-- The @root@ parameter is typically set to @$@ when there is no root object.
+formatPath :: String -> JSONPath -> String
```

Comment:
What about dropping `root` and always setting it to `$`?

Reply:
I thought it would be nice to also give users a way to construct:
- A path referencing a named root object (since `$` is just for an anonymous root object)
- A relative path which could be appended to another path (achieved by setting the root to an empty string)

Some other implementation options:
- Two separate functions, `formatPath` for an absolute path (i.e. include `$` prefix) and `formatPathRelative` for a relative path (i.e. empty string for root, users free to prefix a root in their own code)
- Just make users drop the `$` prefix themselves if they want a relative path

That's the context behind my original choice, but I'm happy to go with whichever approach you prefer.

Reply:
Thanks! Supporting all of this sounds like a good idea.

I think I'd prefer having `formatPath` and `formatPathRelative` (or perhaps `formatRelativePath`?). At least from my use cases I've always wanted the `$` so it would be a bit more cumbersome to have to pass `"$"` as the first argument

Reply:
Hmm or is there a better name than`Relative`? I guess it's not accurate if you want to use it for a named root object?

Reply:
I think `Relative` is the right term as the JSON path is relative to some object, which may be the root object or may be an object referenced by another JSON path. I'll go with your suggestion of `formatRelativePath`.